### ISSUE TITLE
customfield_*

### ISSUE THREAD

**Issue by raikakhodadad1** [free/jiralert#8]
Hi, 

Do you have an example of a selectlist or multiselect list customfield working? What would the format be in jiralert.yml? I have tried to input the customfield multiple ways however wasn't able to get it working. See examples below what I tried:

1) 
```yaml
receivers:
  - name: 'jira-ab'
    project: TOSD 
    components: [ 'RMA' ]
    add_group_labels: false
    fields:
      customfield_10863: DCOPS
```
2) 
```yaml
receivers:
  - name: 'jira-ab'
    project: TOSD 
    components: [ 'RMA' ]
    add_group_labels: false
    fields:
      customfield_10863: {"value": "DCOPS"}
```
3) 
```yaml
receivers:
  - name: 'jira-ab'
    project: TOSD 
    components: [ 'RMA' ]
    add_group_labels: false
    fields:
      customfield_10863: |
        {"value": "DCOPS"}
```
4)
```yaml
receivers:
  - name: 'jira-ab'
    project: TOSD 
    components: [ 'RMA' ]
    add_group_labels: false
    fields:
      customfield_10863: |
        "value": "DCOPS"
```

**Comment by free** [comment]
What exactly was the result of your various attempts? I.e. what was the log output from JIRAlert, was an issue created but without the custom field or did the creation fail entirely etc.

**Comment by raikakhodadad1** [comment]
Sorry for not providing the error message. So the error message is : 

E0516 15:35:24.939902   32653 main.go:131] 500 Internal Server Error: err=JIRA request https://contegixapp1.livenation.com/jira/rest/api/2/issue/ returned status 400 , body "{\"errorMessages\":[],\"errors\":{\"customfield_10863\":\"Could not find valid 'id' or 'value' in the Parent Option object.\"}}" receiver="jira-ab" groupLabels=map[alertname:TestAlertRaika]

And from the testing I've done the error above is coming from Jira which basically means Jira is unable to parse out the value for the custom field because it's not in the correct format. When my colleague and I were trying to troubleshoot we realized the field is parsing out in a different formatting that expected from jira. For example for this format 
customfield_10863: |
        "value": "DCOPS"

it would print out : 
"customfield_10863" : "{"value": "DCOPS"}" 

This is as close as we were able to get it to the jira accepted formatting which is : 
"customfield_11167": { "value": "Troubleshooting" }

And if we did not pass it in as a string aka the first case I mentioned above it was printing out something like so with brackets (if I recall correctly) : 

customfield_10863" : [value: "DCOPS"]

So basically the formatting at the end is a bit off for the Unknows fields. But also there needs to be a way to handle arrays for custom fields as well with acceptable formatting of : 
"customfield_19802": [{ "value": "TechOps CloudSys" }]

which I also don't believe is currently being handled. 

My colleague was able to put in a change that seems to have fixed the issue for non-array custom fields (https://github.com/csquire/jiralert/commit/) but we are still testing it before we did a pull request :)

**Comment by free** [comment]
I am obviously not a heavy JIRA user, so I didn't know how custom fields might look like. I had assumed the values would always be plain strings. Had I scrolled down the JIRA API examples link I intended to provide in my previous comment (which I've edited to add the link now), I would have seen [a whole collection of custom field types and their JSON representation](https://developer.atlassian.com/server/jira/platform/jira-rest-api-examples/#setting-custom-field-data-for-other-field-types).

I've just pushed [this commit](https://github.com/free/jiralert/commit/562cc1f37072c9127bdb7c968954ba5979a7cdd8), which should allow for custom field values to be any combination of `map` (with string keys only), `array`, `string`, `number`, `bool`, `null`. Meaning that your examples (2), (3) and (4) should all work now. It also executes any templates present in string keys or values (recursively). Meaning a value of `"{{ .Status }}"` should be sent to JIRA as `"firing"`.

Please try it out and let me know how it goes.

**Comment by free** [comment]
Oh and thank you @csquire for figuring out where the issue was and providing a first version of the fix.

**Comment by charlspjohn** [comment]
My issue resolved by using:
`fields:
  customfield_xxxx: [{"value":"Not a system related task"}]`

Menu:
![image](https://user-images.githubusercontent.com/14850736/40266853-1ccda07e-5b70-11e8-8313-bb693d9abb02.png)

**Comment by free** [comment]
Closing as fixed, let me know if you still have problems. Also added a few examples and a link to JIRA documentation examples to the config file.

All part of [release 0.6](https://github.com/free/jiralert/releases/tag/0.6).

**Comment by wa-lker-dev** [comment]
If the custom fields value varies between invocations is that supposed to be updated when a new alert triggers? I've noticed that perhaps this doesn't happen and was wondering if it is only set on the initial alert.
